gnome-initial-setup/gnome-initial-setup.c

```diff
@@ -339,10 +359,16 @@ main (int argc, char *argv[])
    * dont have a normal user session and need to initialize
    * the keyring manually so that we can pass the credentials
    * along to the new user in the handoff.
+   * We also need to connect to GDM so it instructs plymouth to quit,
+   * otherwise it stays around until we reach the last (summary) page.
    */
-  if (mode == GIS_DRIVER_MODE_NEW_USER)
+  if (mode == GIS_DRIVER_MODE_NEW_USER) {
     gis_ensure_login_keyring ();
 
+    if (!connect_to_gdm ())
```

Comment:
I don't think this is the right place to do this. The `summary` page will ultimately connect to GDM, so I think it'd be better to move the call to its `connect_to_gdm` function into `gis_summary_page_constructed`, and stash the `GdmGreeter` and `GdmUserVerifier` in its `Private` structure.

Reply:
That makes much more sense. I'm going with that approach (which invalidates the previous comments).